lisa/sut_orchestrator/azure/platform_.py

```diff
@@ -1423,6 +1428,35 @@ def _create_node_arm_parameters(
         if capability.network_interface.data_path == schema.NetworkDataPath.Sriov:
             arm_parameters.enable_sriov = True
 
+        # Convert Security Profile Setting to Arm Parameter
+        _security_profile_mapping = {
```

Comment:
The logic is better in the feature than platform. The arm_parameter is passed into the before_deploy as well. It can be changed in feature too.

Reply:
Please correct me if I am wrong.
My consideration was that security profile should be 'per node' not 'per platform'.
If a runbook wanted to specify multiple nodes with different security profile feature settings, this function will run for every node but if I put the code in before_deployment it will only run once and only only security profile feature settings will be provided.

Is this a case I should be considering?

Reply:
I have moved the logic to on_before_deployment.
I did not use the FeatureSettings, instead using the environment node capabilities to build the arm parameters.
Please make sure all my assumptions look safe, for example environment.nodes should always be equal length to arm_parameters.nodes.

Reply:
Yes, they are equal. The logic looks not organized, but provides max flexibility.